### PR TITLE
IAAM-279 - wrap connectionPoolOptions logic in try/catch to prevent u…

### PR DESCRIPTION
**Pull request by garyburgmann**
…ncaught  PDOException leaking credentials

### CODE REVIEW

File: DB.php [hunk 1]
```diff
@@ -57,15 +57,23 @@ public static function connectionOptions(string $name, $forceSlave = false, $for
...
+            ];
+            $o['pdo'] = new $pdo(
+                "mysql:host={$o['host']};dbname={$o['dbname']};port={$o['port']}",
+                $o['user'],
+                $o['password'],
+                $pdoOpions
+            );
+            return $o;
+        } catch (\Exception | \PDOException $e) {
+            throw new \Exception('DB::connectionPoolOptions error: ' . $e->getMessage());
```
Comment (shangzhipan) [inline DB.php:75]: I would not change the type of Exception that is thrown here. If it is a PDOException, keep throwing a PDOException. We are not sure what the upstream services are catching.

Comment (garyburgmann) [reply DB.php:75]: @shangzhipan I don't pretend to be a PHP expert, so I was just trying to follow the best practice as outlined here:

https://www.php.net/manual/en/class.pdoexception.php -> "You should not throw a PDOException from your own code"

I get your point though. The problem here is the uncaught exception, so we are changing a PHP Fatal error -> Exception:

https://app.datadoghq.com/logs?query=env%3Ak8s-qa%20service%3Aportal&event=AQAAAYBEGD8N3Ntz-QAAAABBWUJFR0VLS0FBQkFVQ1RyZzJvQW9BQUE&index=&from_ts=1650406301322&to_ts=1650409901322&live=false

```sh
PHP message: PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000] [9002] SSL connection is required. Please specify SSL options and retry. in /app/vendor/go1/util_core/DB.php:66
```

And given that this behavior is part of the PHP bootstrapping, I think we should be safe from affecting upstream services.

But your call. Just trying to give you the context and explain that not throwing PDOException was for a reason :)

Comment (sanglt) [reply DB.php:75]: In this case, I prefer to extend the [PDOException](https://www.php.net/manual/en/language.exceptions.extending.php). So the upstream can still catch the PDOException.

Comment (garyburgmann) [reply DB.php:75]: Fair enough. The leak of credentials only comes from the uncaught `PDOException`, so catching and bubbling up the exception should still prevent the leak.

Comment (garyburgmann) [reply DB.php]: latest commit just throws `$e`

Comment (shangzhipan) [reply DB.php]: @garyburgmann This won't change anything if you just catch the exception and throw the same one again. If I see it correctly the credentials are logged in the stacktrace. Duy found a nice article here: https://chriswhite.is/coding/hiding-sensitive-arguments-in-php-stack-traces/ Please have a look.

Comment (garyburgmann) [reply DB.php]: I see. My lack of PHP fu on full display!

@sanglt is there a reason in our PHP images for:

```sh
$ docker run --rm go1com/php:7-nginx php -r 'echo ini_get("display_errors") . "\n";'
1
```

**Pull request merged**